lib/ui/screens/album_screen.dart

```diff
@@ -42,195 +86,435 @@ class _AlbumScreenState extends State<AlbumScreen> {
-            content: Text('Could not share the album.'),
-          ));
-        }
-      },
-    );
-  }
+  Widget _smallGallery(List<AlbumPhoto> photos) => Scrollbar(
+          child: GridView.builder(
+        key: const PageStorageKey('album'),
+        gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+          crossAxisSpacing: 8,
+          mainAxisSpacing: 8,
+          crossAxisCount: 3,
+        ),
+        itemCount: photos.length,
+        physics: const AlwaysScrollableScrollPhysics(),
+        padding: const EdgeInsets.all(8),
+        itemBuilder: (context, index) => _makePhotoCard(
+          photos,
+          index,
+        ),
+      ));
 
   @override
   Widget build(BuildContext context) {
     return BlocBuilder<AlbumCubit, AlbumState>(
       bloc: _albumCubit,
+      //TODO: maybe make this a different function?
       builder: (context, state) {
-        if (state.hasException) {
-          return Scaffold(
-            appBar: ThaliaAppBar(
-              title: Text(widget.album?.title.toUpperCase() ?? 'ALBUM'),
-              actions: [_makeShareAlbumButton(widget.slug)],
-            ),
-            body: ErrorScrollView(state.message!),
-          );
-        } else if (state.isLoading) {
-          return Scaffold(
+        Widget mainScaffold = Scaffold(
             appBar: ThaliaAppBar(
-              title: Text(widget.album?.title.toUpperCase() ?? 'ALBUM'),
-              actions: [_makeShareAlbumButton(widget.slug)],
+              title: Text(state.result?.title.toUpperCase() ??
+                  widget.album?.title.toUpperCase() ??
+                  'ALBUM'),
+              actions: [_makeShareAlbumButton(context, widget.slug)],
             ),
-            body: const Center(child: CircularProgressIndicator()),
-          );
-        } else {
-          return Scaffold(
-            appBar: ThaliaAppBar(
-              title: Text(state.result!.title.toUpperCase()),
-              actions: [_makeShareAlbumButton(widget.slug)],
+            body: state.hasException
+                ? ErrorScrollView(state.message!)
+                : state.isLoading
+                    ? const Center(child: CircularProgressIndicator())
+                    : _smallGallery(state.result!.photos));
+
+        if (galleryShown && !state.hasException && !state.isLoading) {
+          Album album = state.result!;
+          List<bool> likedlist = album.photos.map((e) => e.liked).toList();
+          List<int> likeslist = album.photos.map((e) => e.numLikes).toList();
+
+          // We change the pagecontroler with a new initialPage because
+          // it is impossible to change the initial page after it has been created.
+          // We cannot jump to the page because it is not attached jet. When it opens
+          // the gallery it will use the initialPage instead of last jumped-to page.
+          mainPageController = PageController(initialPage: initialGalleryIndex);
```

Comment:
I think it should be possible to do this from `didUpdateDependencies`? I'm not sure though

Reply:
I am not sure how, this method is only called right after initstate. If you have an idea on how this would work maybe show me tomorrow